docs/Modules/ecdsa/ecdsasmartcontracts.mdx

```diff
@@ -0,0 +1,85 @@
+---
+sidebar_label: "Smart Contract Deep Dive"
+sidebar_position: 2
+description: "In-depth smart contract insights into Biconomy's ECDSA Ownership Registry Module."
+---
+
+# 📜 Smart Contract Deep Dive 🛠️
+
+## Introduction
+
+This document delves into the `EcdsaOwnershipRegistryModule` for Biconomy Smart Accounts, focusing on key functionalities and security aspects.
+
+## Core Functionalities
+
+### User Operation Validation (`validateUserOp`) 🛡️
+
+```solidity
+// Validates user operations signed by an EOA
+function validateUserOp(UserOperation calldata userOp, bytes32 userOpHash)
+    external view returns (uint256) {
+    (bytes memory cleanEcdsaSignature, ) = abi.decode(userOp.signature, (bytes, address));
```

Comment:
code snippets would require updates after we merge gas optimizations PR on SA repo

>https://github.com/bcnmy/scw-contracts/blob/127bb321d03b02fc1f7b21f4278e3eb10452629c/contracts/smart-account/modules/EcdsaOwnershipRegistryModule.sol#L77-L82

Reply:
maybe change it now? as changes should be merged today

Reply:
done